### DISCUSSION TITLE
Capacitor v3 deprecated handlers (back button) with Ionic

### DISCUSSION THREAD

**Discussion by theorzr** [ionic-team/capacitor#4576] [Q&A] [answered]
I'm currently testing the Capacitor v3 (3.0.0-rc.3) release candidate because it seems really promising, and the new plugin API is way better.
But I was quite disappointed when I saw that you deprecated some handlers such as the back button (marked as "legacy handlers", [here](https://github.com/ionic-team/capacitor/blob/0429905efcf69097e7759b8d21d74459b8dd42c5/android/capacitor/src/main/assets/native-bridge.js#L103), the event is not longer triggered since #4067).

This event was used by Ionic to provide an intuitive navigation but this no longer works.

How can I achieve this using capacitor v3, should I implement this behavior in a custom plugin ?

**Comment by thomasvidas** [comment, marked as answer]
You can use the `@capacitor/app` plugin and implement a `BackButtonListener` using this code (docs [here](https://capacitorjs.com/docs/apis/app#addlistenerbackbutton))

```typescript
const handle: PluginListenerHandle = App.addListener('backButton', listenerFunc);
```

The reason this was changed was because for v3 was that we pulled out all of the plugins so they would be imported dynamically rather than bundling the plugins in with the Capacitor core runtime

**Reply by theorzr** [reply]
You can consider this question answered! I already got the answer on another issue but thank you anyway!
